File: Assignment4/Assignment4/Logic/CalendarGenerator.swift
import Foundation

class CalendarGenerator {
  func generate(yearAndMonth: DateComponents) -> [[Int]]  {
      var cal = [[Int]](repeating: [Int](repeating: 0, count: 7), count: 6)
      var prevmon = yearAndMonth.month! - 1
      var year = yearAndMonth.year!
      var days = 0
      var prevdays = 0
      var before = 0
      
      
      var components = DateComponents()
      components.year = yearAndMonth.year
      components.month = yearAndMonth.month
      components.day = 1
      
      
      if (prevmon == 0) {
        prevmon = 12
        year -= 1
      }
      
      
      if let date = Calendar.current.date(from: components) {
          let range = Calendar.current.range(of: .day, in: .month, for: date)
          days = range!.count
          before = Calendar.current.component(.weekday, from: date)
      }
      
      let prevdate = DateComponents(year: year, month: prevmon)
      
      if let date = Calendar.current.date(from: prevdate) {
          let range = Calendar.current.range(of: .day, in: .month, for: date)
          prevdays = range!.count
      }
      
      var counter = prevdays - before + 2
      if (before == 1) {
        counter = 1
      }
      
      for i in 0..<7 {
        cal[0][i] = counter
        counter += 1
        if (counter > prevdays) {
          counter = 1
        }
      }
      
      for i in 1..<6 {
        for j in 0..<7 {
          cal[i][j] = counter
          counter += 1
          if (counter > days) {
            counter = 1
          }
        }
      }
      
      print(cal)
      
          return cal
  }
}

/*
 /*
 #######################################################################
 #
 # Copyright (C) 2022-2023 David C. Harrison. All right reserved.
 #
 # You may not use, distribute, publish, or modify this code without
 # the express written permission of the copyright holder.
 #
 #######################################################################
 */

 import java.time.YearMonth

 class CalendarArray {
   /**
    * See assignment specification.
    */



   fun generate(month: YearMonth): Array<IntArray> {


     val cal = Array(6) { IntArray(7) }
     var prevmon = month.month.value - 1
     var year = month.year
     if (prevmon == 0) {
       prevmon = 12
       year -= 1
     }
     val prev = YearMonth.of(year, prevmon)
     val days = month.lengthOfMonth()
     val prevdays = prev.lengthOfMonth()
     val before = month.atDay(1).dayOfWeek.value % 7



     var counter = prevdays - before + 1
     if (before == 0) {
       counter = 1
     }
     println(counter)
     // var cal = Array(6, { IntArray(7) })
     for (i in 0..6) {
       cal[0][i] = counter
       counter += 1
       if (counter > prevdays) {
         counter = 1
       }
     }
     for (i in 1..5) {
       for (j in 0..6) {
         cal[i][j] = counter
         counter += 1
         if (counter > days) {
           counter = 1
         }
       }
     }
    /* for (x in cal){
       for (y in x){
         print(y)
         print(" ")
       }
       println()
     }*/
     return cal
   }
 }
 */

File: Assignment7/src/Logic/CalendarGenerator.js
/*
 * Copyright (C) 2022-2023 David C. Harrison. All right reserved.
 *
 * You may not use, distribute, publish, or modify this code without
 * the express written permission of the copyright holder.
 */

export default class {
  /**
   * @param {object} date
   * @return {[[Number]]} 6x7 array of integers
   */

  static generate(date) {
    console.log('Date');
    let cal = [];
    let month = date.getMonth();
    console.log('Month');
    console.log(month);
    let monGen = month + 1;
    let year = date.getFullYear();
    console.log('Year');
    console.log(year);
    let GetFirst = new Date(year, month, 1);

    let before = GetFirst.getDay();

    console.log('Day of the week');
    console.log(before);

    let prevyear = year;
    let prevmonth = date.getMonth() - 1;

    if (monGen > 11) {
      monGen = 0;
      year = year + 1;
    }
    let curr = new Date(year, monGen, 0);
    let days = curr.getDate();
    if (prevmonth <= 0) {
      prevmonth = 11;
      prevyear = year - 1;
      console.log('Jan');
    }

    let PrevGen = prevmonth + 1;

    if (PrevGen > 11) {
      PrevGen = 0;
      prevyear = prevyear + 1;
    }

    let prevdate = new Date(prevyear, PrevGen, 0);
    let prevdays = prevdate.getDate();
    console.log('Prevdays');
    console.log(prevdays);

    let counter = prevdays - before + 1;
    if (before === 0) {
      counter = 1;
    }

    console.log('counter');
    console.log(counter);
    for (let i = 0; i < 6; i++) {
      cal[i] = [];
    }

    for (let i = 0; i < 7; i++) {
      cal[0][i] = counter;
      counter += 1;
      if (counter > prevdays) {
        counter = 1;
      }
    }

    for (let i = 1; i < 6; i++) {
      for (let j = 0; j < 7; j++) {
        cal[i][j] = counter;
        counter += 1;
        if (counter > days) {
          counter = 1;
        }
      }
    }

    return cal;
  }
}
